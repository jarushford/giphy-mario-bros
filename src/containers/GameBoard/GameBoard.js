import React from 'react'
import Loading from '../../components/Loading/Loading'
import PlayerScore from '../../components/PlayerScore/PlayerScore'
import { connect } from 'react-redux'
import { uid } from 'react-uid';
import '../../main.scss'

export function GameBoard({ players, judge, caption, currentPlayer }) {
  if (!players.length) {
    return <Loading />
  }

  const scores = players.map(player => {
    if (player.player === judge) {
      return <PlayerScore {...player} key={uid(player)} active={true} />
    }
    return <PlayerScore {...player} key={uid(player)} active={false} />
  })

  let instruction
  if (currentPlayer === judge) {
    instruction = <h2>Player {currentPlayer}, choose a winner</h2>
  } else {
    instruction = <h2>Player {currentPlayer}, choose a GIF</h2>
  }

  return (
    <section className="game-board">
      <h1 className="title">Giphy Mario Bros.</h1>
      <div className="player-scores">{scores}</div>
      <h3>{caption[0]}</h3>
      {instruction}  
    </section>
  )
}

export const mapStateToProps = (state) => ({
  players: state.players,
  judge: state.judge,
  currentPlayer: state.currentPlayer,
  caption: state.caption
})

export default connect(mapStateToProps)(GameBoard)