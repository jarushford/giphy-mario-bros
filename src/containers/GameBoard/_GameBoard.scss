.game-board {
  @include main();
  cursor: default;
  min-width: 700px;
  margin-top: 70px;

  h3 {
    text-align: center;
    color: $white;
    background: $red;
    padding: 10px;
    border-radius: 4px;
    font-family: 'Viga', sans-serif;
    margin: 10px auto 30px;
  }

  h2 {
    text-align: center;
    color: $white;
    font-family: 'Mario';
    text-shadow: $outlinethin;
    margin: 25px auto 5px;
  }
}

.player-scores {
  display: flex;
  justify-content: space-around;
}

@media screen and (max-width: 800px) {
  .game-board {
    min-width: unset;
    width: 100%;
  }
}

@media screen and (max-width: 600px) {
  .player-scores {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .game-board {
    width: 80%;
  }
}