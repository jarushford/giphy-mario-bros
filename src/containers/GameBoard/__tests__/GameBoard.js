import React from 'react'
import { GameBoard, mapStateToProps } from '../GameBoard'
import { shallow } from 'enzyme'

describe('GameBoard', () => {
  let mockPlayers
  let mockJudge
  let mockCaption
  let mockCurrent

  beforeEach(() => {
    mockPlayers = [{ player: 1, score: 0 }, { player: 2, score: 0 }, { player: 3, score: 0 } ]
    mockJudge = 2
    mockCaption = ['Test Caption']
    mockCurrent = 3
  })

  it('should match the snapshot', () => {
    const wrapper = shallow(<GameBoard players={mockPlayers} judge={mockJudge} caption={mockCaption} currentPlayer={mockCurrent} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if the current player is the judge', () => {
    const wrapper = shallow(<GameBoard players={mockPlayers} judge={mockJudge} caption={mockCaption} currentPlayer={2} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with no players', () => {
    const wrapper = shallow(<GameBoard players={[]} judge={mockJudge} caption={mockCaption} currentPlayer={mockCurrent} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return a props object with the correct properties from state', () => {
      const mockState = {
        players: [{ player: 1, score: 0 }, { player: 2, score: 0 }, { player: 3, score: 0 } ],
        judge: 1,
        currentPlayer: 2,
        caption: ['Test Caption'],
        error: '',
        choices: []
      }
      const expected = {
        players: [{ player: 1, score: 0 }, { player: 2, score: 0 }, { player: 3,score: 0 } ],
        judge: 1,
        currentPlayer: 2,
        caption: ['Test Caption']
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})