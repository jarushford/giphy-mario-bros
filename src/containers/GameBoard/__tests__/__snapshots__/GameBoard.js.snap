// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameBoard
    caption={
      Array [
        "Test Caption",
      ]
    }
    currentPlayer={3}
    judge={2}
    players={
      Array [
        Object {
          "player": 1,
          "score": 0,
        },
        Object {
          "player": 2,
          "score": 0,
        },
        Object {
          "player": 3,
          "score": 0,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Giphy Mario Bros.
        </h1>,
        <div
          className="player-scores"
        >
          <PlayerScore
            active={false}
            player={1}
            score={0}
          />
          <PlayerScore
            active={true}
            player={2}
            score={0}
          />
          <PlayerScore
            active={false}
            player={3}
            score={0}
          />
        </div>,
        <h3>
          Test Caption
        </h3>,
        <h2>
          Player 
          3
          , choose a GIF
        </h2>,
      ],
      "className": "game-board",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Giphy Mario Bros.",
          "className": "title",
        },
        "ref": null,
        "rendered": "Giphy Mario Bros.",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <PlayerScore
              active={false}
              player={1}
              score={0}
            />,
            <PlayerScore
              active={true}
              player={2}
              score={0}
            />,
            <PlayerScore
              active={false}
              player={3}
              score={0}
            />,
          ],
          "className": "player-scores",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "uid1",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "player": 1,
              "score": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "uid2",
            "nodeType": "function",
            "props": Object {
              "active": true,
              "player": 2,
              "score": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "uid3",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "player": 3,
              "score": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Caption",
        },
        "ref": null,
        "rendered": "Test Caption",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Player ",
            3,
            ", choose a GIF",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Player ",
          3,
          ", choose a GIF",
        ],
        "type": "h2",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Giphy Mario Bros.
          </h1>,
          <div
            className="player-scores"
          >
            <PlayerScore
              active={false}
              player={1}
              score={0}
            />
            <PlayerScore
              active={true}
              player={2}
              score={0}
            />
            <PlayerScore
              active={false}
              player={3}
              score={0}
            />
          </div>,
          <h3>
            Test Caption
          </h3>,
          <h2>
            Player 
            3
            , choose a GIF
          </h2>,
        ],
        "className": "game-board",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Giphy Mario Bros.",
            "className": "title",
          },
          "ref": null,
          "rendered": "Giphy Mario Bros.",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <PlayerScore
                active={false}
                player={1}
                score={0}
              />,
              <PlayerScore
                active={true}
                player={2}
                score={0}
              />,
              <PlayerScore
                active={false}
                player={3}
                score={0}
              />,
            ],
            "className": "player-scores",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "uid1",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "player": 1,
                "score": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "uid2",
              "nodeType": "function",
              "props": Object {
                "active": true,
                "player": 2,
                "score": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "uid3",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "player": 3,
                "score": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Caption",
          },
          "ref": null,
          "rendered": "Test Caption",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Player ",
              3,
              ", choose a GIF",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Player ",
            3,
            ", choose a GIF",
          ],
          "type": "h2",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`GameBoard should match the snapshot if the current player is the judge 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameBoard
    caption={
      Array [
        "Test Caption",
      ]
    }
    currentPlayer={2}
    judge={2}
    players={
      Array [
        Object {
          "player": 1,
          "score": 0,
        },
        Object {
          "player": 2,
          "score": 0,
        },
        Object {
          "player": 3,
          "score": 0,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Giphy Mario Bros.
        </h1>,
        <div
          className="player-scores"
        >
          <PlayerScore
            active={false}
            player={1}
            score={0}
          />
          <PlayerScore
            active={true}
            player={2}
            score={0}
          />
          <PlayerScore
            active={false}
            player={3}
            score={0}
          />
        </div>,
        <h3>
          Test Caption
        </h3>,
        <h2>
          Player 
          2
          , choose a winner
        </h2>,
      ],
      "className": "game-board",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Giphy Mario Bros.",
          "className": "title",
        },
        "ref": null,
        "rendered": "Giphy Mario Bros.",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <PlayerScore
              active={false}
              player={1}
              score={0}
            />,
            <PlayerScore
              active={true}
              player={2}
              score={0}
            />,
            <PlayerScore
              active={false}
              player={3}
              score={0}
            />,
          ],
          "className": "player-scores",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "uid4",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "player": 1,
              "score": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "uid5",
            "nodeType": "function",
            "props": Object {
              "active": true,
              "player": 2,
              "score": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "uid6",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "player": 3,
              "score": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Caption",
        },
        "ref": null,
        "rendered": "Test Caption",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Player ",
            2,
            ", choose a winner",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Player ",
          2,
          ", choose a winner",
        ],
        "type": "h2",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Giphy Mario Bros.
          </h1>,
          <div
            className="player-scores"
          >
            <PlayerScore
              active={false}
              player={1}
              score={0}
            />
            <PlayerScore
              active={true}
              player={2}
              score={0}
            />
            <PlayerScore
              active={false}
              player={3}
              score={0}
            />
          </div>,
          <h3>
            Test Caption
          </h3>,
          <h2>
            Player 
            2
            , choose a winner
          </h2>,
        ],
        "className": "game-board",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Giphy Mario Bros.",
            "className": "title",
          },
          "ref": null,
          "rendered": "Giphy Mario Bros.",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <PlayerScore
                active={false}
                player={1}
                score={0}
              />,
              <PlayerScore
                active={true}
                player={2}
                score={0}
              />,
              <PlayerScore
                active={false}
                player={3}
                score={0}
              />,
            ],
            "className": "player-scores",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "uid4",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "player": 1,
                "score": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "uid5",
              "nodeType": "function",
              "props": Object {
                "active": true,
                "player": 2,
                "score": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "uid6",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "player": 3,
                "score": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Caption",
          },
          "ref": null,
          "rendered": "Test Caption",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Player ",
              2,
              ", choose a winner",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Player ",
            2,
            ", choose a winner",
          ],
          "type": "h2",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`GameBoard should match the snapshot with no players 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameBoard
    caption={
      Array [
        "Test Caption",
      ]
    }
    currentPlayer={3}
    judge={2}
    players={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
